### PR TITLE
feat(kuma-cp) validate tags and selectors

### PR DESCRIPTION
**Pull request by lobkovilya**
### Summary

That PR limits the character set that could be used for tag's names and values. The same restriction is done for selectors as well. 

Tags could consist of alphanumeric chars, dots, dashes, underscores, and colon. The last one is needed for service names with ports, e.g `backend:8080`.

### CODE REVIEW

File: pkg/core/resources/apis/mesh/dataplane_validator.go [hunk 1]
```diff
@@ -134,9 +130,21 @@ func validateGateway(gateway *mesh_proto.Dataplane_Networking_Gateway) validator
...
+
+func validateTags(tags map[string]string) validators.ValidationError {
+	var result validators.ValidationError
+	for name, value := range tags {
 		if value == "" {
-			result.AddViolationAt(validators.RootedAt("tags").Key(name), `tag value cannot be empty`)
+			result.AddViolationAt(validators.RootedAt("tags").Key(name), `value cannot be empty`)
+		}
+		if !nameCharacterSet.MatchString(name) {
+			result.AddViolationAt(validators.RootedAt("tags").Key(name), `key must consist of alphanumeric characters, dots, dashes and underscores`)
```
Comment (jakubdyszkiewicz) [inline pkg/core/resources/apis/mesh/dataplane_validator.go:144]: how about `tag name` instead of `key`

Comment (jakubdyszkiewicz) [reply pkg/core/resources/apis/mesh/dataplane_validator.go:144]: same with value

File: pkg/core/resources/apis/mesh/dataplane_validator.go [hunk 2]
```diff
@@ -134,9 +130,21 @@ func validateGateway(gateway *mesh_proto.Dataplane_Networking_Gateway) validator
...
+		}
+		if !nameCharacterSet.MatchString(value) {
```
Comment (jakubdyszkiewicz) [inline pkg/core/resources/apis/mesh/dataplane_validator.go:146]: You also allow `*` here. I think dataplane value tag should have separate regex without `*`

File: pkg/core/resources/apis/mesh/validators.go [hunk 3]
```diff
@@ -150,3 +151,22 @@ func ProtocolValidator(protocols ...string) SelectorValidatorFunc {
 		return
 	}
 }
+
+var nameCharacterSet = regexp.MustCompile(`^[a-zA-Z0-9\.\-_:]*$`)
+var selectorCharacterSet = regexp.MustCompile(`^([a-zA-Z0-9\.\-_:]*|\*)$`)
+
+func SelectorCharacterSetValidator(path validators.PathBuilder, selector map[string]string) validators.ValidationError {
```
Comment (jakubdyszkiewicz) [inline pkg/core/resources/apis/mesh/validators.go:158]: To be honest, I'd use this in `ValidateSelector` so it's checked on every policy, not only on fault injection

File: pkg/core/resources/apis/mesh/dataplane_validator.go (hunk 1 above)
Comment (lobkovilya) [reply pkg/core/resources/apis/mesh/dataplane_validator.go:144]: Yes, I wanted to have the same code for Selectors and Tags, so I modified that message and forget to revert it

File: pkg/core/resources/apis/mesh/dataplane_validator.go (hunk 2 above)
Comment (lobkovilya) [reply pkg/core/resources/apis/mesh/dataplane_validator.go:146]: No, I believe I don't: 
```
var nameCharacterSet = regexp.MustCompile(^[a-zA-Z0-9\.\-_:]*$)
var selectorCharacterSet = regexp.MustCompile(`^([a-zA-Z0-9\.\-_:]*|\*)$`)
```
I was thinking 2 sets are enough, but maybe I should reconsider naming

Comment (jakubdyszkiewicz) [reply pkg/core/resources/apis/mesh/dataplane_validator.go:146]: You are right, sorry. I think it makes sense to have
```
var tagNameCharacterSet = regexp.MustCompile(^[a-zA-Z0-9\.\-_:]*$)
var selectorCharacterSet = regexp.MustCompile(`^([a-zA-Z0-9\.\-_:]*|\*)$`)
var tagValueCharacterSet = regexp.MustCompile(`^([a-zA-Z0-9\.\-_:]*)$`)
```

**Pull request merged**